psqlextra/compiler.py

```diff
@@ -72,19 +123,32 @@ def _does_dict_contain_expression(data: dict) -> bool:
 class PostgresInsertCompiler(SQLInsertCompiler):
     """Compiler for SQL INSERT statements."""
 
-    def __init__(self, *args, **kwargs):
-        """Initializes a new instance of :see:PostgresInsertCompiler."""
+    def as_sql(self, return_id=False):
+        """Builds the SQL INSERT statement."""
+        queries = [
+            (append_caller_to_sql(sql), params)
+            for sql, params in super().as_sql()
+        ]
+
+        return queries
+
 
+class PostgresInsertOnConflictCompiler(SQLInsertCompiler):
```

Comment:
**question**: Why was this renamed and splitted?

Reply:
The original `PostgresInsertCompiler` only applied to `INSERT ... ON CONFLICT` queries and is incompatible with `INSERT` queries without `ON CONFLICT` condition. This has been invisible here and has been covered by `PostgresOperations.compiler` by preferring the Django-detected compiler. The change just made that already existing split visible and added a simple wrapper for plain `INSERT`s.